File: src/Game1/View/GroundStrategy.cs
namespace Game.View
{
    public sealed class GroundStrategy : ITileStrategy
    {
        private readonly TextureHolder texture;

        public GroundStrategy(TextureHolder texture)
        {
            this.texture = texture;
        }
        public bool CanExecute(LocationType[] neighbors)
        {
            return neighbors[Directions.NeighborThis] == LocationType.Ground;
        }

        public TextureHolder Execute(LocationType[] neighbors)
        {
            return texture;
        }
    }
}

File: src/core/src/com/mygdx/game/view/GroundStrategy.java
package com.mygdx.game.view;

public final class GroundStrategy implements ITileStrategy {

    private final TextureHolder texture;

    public GroundStrategy(TextureHolder texture) {
        this.texture = texture;
    }

    public boolean CanExecute(LocationType[] neighbors) {
        return neighbors[Directions.NeighborThis] == LocationType.Ground;
    }

    public TextureHolder Execute(LocationType[] neighbors) {
        return texture;
    }
}
